### ISSUE TITLE
Excluded files, defaults, gitignore and zip archive, the big picture

### ISSUE THREAD

**Issue by naholyr** [byteclubfr/copycast#33]
We have several issues around excluded files today:

* If one wants to cast assets, it's currently impossible. Maybe we should remove our defaults and only rely on .gitignore?
* When one downloads the zip archive, he may expect to have a working copy (assets, node_modules)? But you definitely don't want them to pollute UI? Should we ignore exclusions for zip file (but what if there is a private file amongst them?)?

So I take all that and came to the following conclusions I'd like to discuss:

* Zip file should contain all files, except gitignore'd ones, properly serving the purpose of grabbing a full copy of the project
* UI should only show un-gitignored TEXT files, any binary file should not be visible
  * however we should still send the WHOLE tree to socket, and filter front-side so we can make it an option for users who want to cherry-pick a single binary file
  * we should even watch and notify changes on binary files for the day we want to show a binary file has been modified (default behavior would become a binary file does not show up unless it's been recently modified)
* So basically: no more default exclusions server-side, and a mime-based exclusion client-side
